Mathlib/LinearAlgebra/TensorProduct/RightExactness.lean

```diff
@@ -0,0 +1,648 @@
+/-
+Copyright (c) 2023 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambetr-Loir
+-/
+
+import Mathlib.LinearAlgebra.TensorProduct
+import Mathlib.LinearAlgebra.Quotient
+
+import Mathlib.RingTheory.TensorProduct
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+import Mathlib.RingTheory.Ideal.QuotientOperations
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+
+/-! # Right exactness properties of tensor product
+
+* For linear maps `f : M →ₗ[R] N`  and `g : N →ₗ[R] P`, `Exact f g` says that `range f = ker g``
+
+* `TensorProduct.rTensor_exact` says that one can tensor a short exact sequence on the right,
+  and `rTensor.equiv` gives a LinearEquiv.
+
+* `TensorProduct.lTensor_exact` says that one can tensor a short exact sequence on the left,
+  and `lTensor.equiv` gives a LinearEquiv.
+
+* For `N : Submodule R M`, `LinearMap.exact_subtype_mkQ N` says that
+  the inclusion of the submodule and the quotient map form an exact pair,
+  and `lTensor_mkQ` compute `ker (lTensor Q (N.mkQ))` and similarly for `rTensor_mkQ`
+
+* `TensorProduct.map_ker` computes the kernel of `TensorProduct.map f g'`
+  in the presence of two short exact sequences.
+
+The proofs are those of [bourbaki1989] (chap. 2, §3, n°6)
+
+* Analogue for morphisms of algebras :
+  `Algebra.TensorProduct.ker_map` computes the kernel of `Algebra.TensorProduct.map f g`
+
+* The two more specific lemmas `Algebra.TensorProduct.lTensor_ker`
+  and `Algebra.TensorProduct.rTensor_ker` compute the kernels
+  of `Algebra.TensorProduct.map f id` and `Algebra.TensorProduct.map id g`
+
+## TODO
+
+* Treat the noncommutative case
+
+* Treat the case of modules over semirings
+  (For a possible definition of an exact sequence of commutative semigroups, see
+  [Grillet-1969b], Pierre-Antoine Grillet,
+  *The tensor product of commutative semigroups*,
+  Trans. Amer. Math. Soc. 138 (1969), 281-293, doi:10.1090/S0002-9947-1969-0237688-1 .)
+
+-/
+
+section Exact
+
+variable {M N P : Type*} (f : M → N) (g : N → P)
+
+/-- The maps `f` and `g` form an exact pair :
+  `g y = 0` iff `y` belongs to the image of `f` -/
+def Exact [Zero P] : Prop := ∀ y, g y = 0 ↔ y ∈ Set.range f
```

Comment:
Two questions:

1. Should this definition `Exact` be in the root namespace? (I don't have a good suggestion for a namespace atm, but it feels like the name is too generic to go in root.)
2. Should this definition be in a file called `RightExactness.lean` in a folder on tensor products? It seems likely that files might want to import the definition of exactness without wanting to import tensor products. So I think it makes sense to split this file into two.